graviti/schema/catalog_to_schema.py

```diff
@@ -0,0 +1,339 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022 Graviti. Licensed under MIT License.
+#
+
+"""Functions converting TensorBay Catalog to Graviti Schema."""
+
+from pathlib import Path
+from typing import Any, Callable, Dict, Iterator, List, TypeVar, Union
+
+import filetype
+import yaml
+from tensorbay.dataset import Notes, RemoteData
+from tensorbay.label import (
+    Box2DSubcatalog,
+    Box3DSubcatalog,
+    Catalog,
+    ClassificationSubcatalog,
+    InstanceMaskSubcatalog,
+    Keypoints2DSubcatalog,
+    Label,
+    LabeledBox2D,
+    LabeledKeypoints2D,
+    LabeledMultiPolygon,
+    LabeledMultiPolyline2D,
+    LabeledPolygon,
+    LabeledPolyline2D,
+    MultiPolygonSubcatalog,
+    MultiPolyline2DSubcatalog,
+    PanopticMaskSubcatalog,
+    PolygonSubcatalog,
+    Polyline2DSubcatalog,
+    RLESubcatalog,
+    SemanticMaskSubcatalog,
+    SentenceSubcatalog,
+)
+from tensorbay.label.supports import AttributesMixin, CategoriesMixin, MaskCategoriesMixin
+
+_FILE_TYPE = {
+    "audio": "file.Audio",
+    "image": "file.Image",
+    "point_cloud": "file.PointCloud",
+    "other": "bytes",
+}
+_COMMON_POINT_CLOUD_EXTENSIONS = {"pcd", "ply", "las", "laz"}
+_ATTRIBUTETYPE_TO_SCHEMA_TYPE = {
+    "array": "array",
+    "boolean": "boolean",
+    "integer": "int32",
+    "number": "float32",
+    "string": "string",
+}
+_LABEL_PROCESS = {}
+_Callable = TypeVar("_Callable", bound=Callable[..., Any])
+
+
+def _register(label_type: str, portex_type: str = "") -> Callable[[_Callable], _Callable]:
+    def decorator(func: _Callable) -> _Callable:
+        _LABEL_PROCESS[label_type] = (portex_type, func)
+        return func
+
+    return decorator
+
+
+def _get_categories(subcatalog: Union[CategoriesMixin, MaskCategoriesMixin]) -> Dict[str, Any]:
+    if not hasattr(subcatalog, "categories"):
+        return {}
+
+    fields: Dict[str, Any] = {}
+    if isinstance(subcatalog, MaskCategoriesMixin):
+        fields["categoryIds"] = {
+            category.category_id: category.name for category in subcatalog.categories
+        }
+    else:
+        fields["categories"] = [category.name for category in subcatalog.categories]
+
+    if getattr(subcatalog, "category_delimiter", None):
+        fields["categoryDelimiter"] = subcatalog.category_delimiter
+
+    return fields
+
+
+def _get_attributes(subcatalog: AttributesMixin) -> Dict[str, Any]:
+    if not hasattr(subcatalog, "attributes"):
+        return {}
+
+    attributes: List[Dict[str, Any]] = []
+    for attribute in subcatalog.attributes:
+        info: Dict[str, Any] = {"name": attribute.name}
```

Comment:
what is `info`? what is `field`? what is `field_info`? what is `item`?
we need to reconsider the variable name